lib/amq/settings.rb

```diff
@@ -13,25 +13,35 @@ module Settings
       #
       # @see AMQ::Client::Settings.configure
       def self.default
+        # Consider to move this defaults to client libraries (bunny, amqp, auto-generated amq-protocol client)
```

Comment:
`AMQ::Settings` only holds defaults for the assumed clients and switch between Hash and String like options processing.. In my opinion, all client specific parameters should be handled (validate, fallback to default, other conditional logic, etc.) exactly inside client code, not protocol library.

I'd like to propose to retire client specific options (`frame_max`) form `AMQ::Settings` module and leave only basic ones. "Basic" means all that standard Erlang client supports according to ["URI Specification"](https://www.rabbitmq.com/uri-spec.html) and ["URI query parameters"](https://www.rabbitmq.com/uri-query-parameters.html) official RabbitMQ documentation page.